### ISSUE TITLE
Update Flow 1 to not run jobs 2 and 3 if no cases are returned from job 1

### ISSUE THREAD

**Issue by daissatou2** [OpenFn/unicef-thailand#65]
## Background, context, and business value

Currently, the Phase II Flow 1 implementation runs all three jobs in a flow even if no valid cases are returned from Primero after the first job runs:
1.  ([Flow 1-1 Get Primero Cases [CS]](https://github.com/OpenFn/primero-thailand/blob/staging/jobs/child-shield/getCasesFromPrimero.js), 
2. [Flow 1-2 Get Translations from Primero [CS]](https://github.com/OpenFn/primero-thailand/blob/staging/jobs/child-shield/fetch_translations.js), 
3. [Flow 1-3 Map data & sync to ChildShield [CS]](https://github.com/OpenFn/primero-thailand/blob/staging/jobs/child-shield/syncInterventionsToChildShield.js)) 

See this [parent run](https://www.openfn.org/projects/primero-thailand-io-unicef---staging/runs/0634cc8b-4f66-76c4-8b38-34d7e14e08cb) for example. 0 cases will be synced to CS but flow jobs 2 and 3 still run. This means we are sending unnecessary requests in jobs 2 and 3. 

Is it possible to stop the flow _without throwing an error_ after step 1 if no cases are to be processed? It seems like unnecessary work for the entire flow to run if we know after step 1 no cases need to be processed. 

## The specific request, in as few words as possible

Update this flow to not run jobs 2 and 3 if the length of `filteredCases` is 0 at the end of [job 1](https://github.com/OpenFn/primero-thailand/blob/2bc357f879468e5d01800e7e4229a3056b8af946/jobs/child-shield/getCasesFromPrimero.js#L53). 

## state.json

JOB 1 - Use the manual cursor. You should not see any cases returned from Primero.
```json
{
  "configuration": { "SEE LP Primero Thailand Demo API User (Open Function)" },
  "cursor": "Fetch cases where "last_updated_at" >= ''2022-10-16T00:43:07.000Z'"
}
```

JOB 2:
```json
{
  "configuration": ["See LP MOPH Thailand HIS API - DEMO Alpha (Unicef)'"],
  "data": { "data generated by GET job"},
}
```

JOB 3:
```json
{
  "configuration": ["See LP MOPH Thailand HIS API - DEMO Alpha (Unicef)'"],
  "data": { "final state of last 2 jobs"},
}
```
## adaptor
Job 1: language-primero 
Job 2: language-http
Job 3: language-http

## expression.js
[Flow 1-1 Get Primero Cases [CS]](https://github.com/OpenFn/primero-thailand/blob/staging/jobs/child-shield/getCasesFromPrimero.js) 
[Flow 1-2 Get Translations from Primero [CS]](https://github.com/OpenFn/primero-thailand/blob/staging/jobs/child-shield/fetch_translations.js)
[Flow 1-3 Map data & sync to ChildShield [CS]](https://github.com/OpenFn/primero-thailand/blob/staging/jobs/child-shield/syncInterventionsToChildShield.js)

## Toggl 
`UNICEF Thailand`

**Comment by aleksa-krolls** [comment]
@mtuchi lower priority, but can you take a look at this today and discuss with @taylordowns2000? Not sure what is possible.

**Comment by aleksa-krolls** [comment]
@daissatou2 [this](https://www.openfn.org/projects/primero-thailand-io-unicef---staging/runs/0634d5b7-3c2a-7890-bed1-d6f0025507b1) is what the job 2 and 3 runs look like now if there is no data to sync to Child Shield... no operations are execute and the log just says `Finished`. 

@mtuchi can you please add logs so that if there is NO data to be processed and we're skipping the operations, can you log the message `No data to process; logic in this workflow step is skipped.`

@daissatou2 Looks like this is working well when there is no data to send to Child Shield. But once Mtuchi has implemented the above, please test to make sure that this is still working when we _do_ have data to send to Child Shield. (I want to make sure nothing we had before was broken.)

**Comment by daissatou2** [comment]
@mtuchi please take a look at this [error](https://www.openfn.org/projects/primero-thailand-io-unicef---staging/runs/0634d899-b249-7d21-bbd9-8a67b4595418) I am seeing when data _is_ returned from Primero. 

Job 1 manual cursor which returns data: `'2022-09-28T00:43:07.000Z'`
Job 1 manual cursor which shouldn't return data: `'2022-10-17T00:43:07.000Z'`

**Comment by aleksa-krolls** [comment]
@mtuchi I reverted the J2 job on platform back to [this prior commit](https://github.com/OpenFn/primero-thailand/commit/61518c8d0af607b13607c738afdfca816f45ff16) and the J3 job back to [this commit](https://github.com/OpenFn/primero-thailand/commit/89e3c37b316597771c98c5724ff05b60fd6d40a3) from before we started making changes. Client testing begins tomorrow, so I don't want to block them. 
Please make sure these `noop` related changes are working with both of the Job 1 cursors Aicha provided (see here comment above). When tested and working, please submit a PR to staging for me to review/test before I merge with staging.

**Comment by mtuchi** [comment]
> @mtuchi I reverted the J2 job on platform back to [this prior commit](https://github.com/OpenFn/primero-thailand/commit/61518c8d0af607b13607c738afdfca816f45ff16) and the J3 job back to [this commit](https://github.com/OpenFn/primero-thailand/commit/89e3c37b316597771c98c5724ff05b60fd6d40a3) from before we started making changes. Client testing begins tomorrow, so I don't want to block them. Please make sure these `noop` related changes are working with both of the Job 1 cursors Aicha provided (see here comment above). When tested and working, please submit a PR to staging for me to review/test before I merge with staging.

@aleksa-krolls I have fixed the `noop` undefined error, #69

**Comment by aleksa-krolls** [comment]
@mtuchi when I run J1 with the cursor `2022-09-28T00:43:07.000Z`, J1 and J2 run successfully - [see run](https://www.openfn.org/projects/primero-thailand-io-unicef---staging/runs/0634e66f-7eb6-7a41-a53b-228e47ef2048). 
But J3 fails - see run: https://www.openfn.org/projects/primero-thailand-io-unicef---staging/runs/0634e670-ac0d-7f27-8b25-6f4d08e19484

Unfortunately on platform I cannot see the detailed error message. The logs look like this below. Can you please enhance the logging so that the full error object is printed? 
```
data: { error: [Object] }
  },
  isAxiosError: true,
  toJSON: [Function: toJSON]
```
Can you please sure you're testing locally with the cursor? `2022-09-28T00:43:07.000Z` (If you did, then please run the test a couple of times locally to confirm that re-tries don't throw any errors as well.)

**Comment by aleksa-krolls** [comment]
@mtuchi moving this to Review/QA until we hear back from the client regarding this 500 error.

**Comment by daissatou2** [comment]
This 500 error was caused by bad test data. I've removed that case and this looks good now: https://www.openfn.org/projects/primero-thailand-io-unicef---staging/runs/0634e907-2002-75da-be60-812347bfe8b0
